src/Search/Queue/WatchableQueue.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace eLife\Search\Queue;
+
+interface WatchableQueue
+{
+    /**
+     * Adds item to the queue.
+     *
+     * Mock: Add item to queue.
+     * SQS: This will set the queue item into the memory slot for re-processing.
+     */
+    public function enqueue(QueueItem $item) : bool;
+
+    /**
+     * Starts process of removing item.
+     *
+     * Mock: Move to separate "in progress" queue.
+     * SQS: this will change the timeout of the in-memory item.
+     */
+    public function dequeue(int $timeout = 30) : QueueItem;
+
+    /**
+     * Commits to removing item from queue, marks item as done and processed.
+     *
+     * Mock: Remove item completely.
+     * SQS: this will delete the item from the queue.
+     */
```

Comment:
Looks like it's configuration for the daemon, the API stays the same. We would use RDS in production/end2end.